pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/GrpcBrokerRequestHandler.java

```diff
@@ -0,0 +1,198 @@
+      BrokerMetrics brokerMetrics, TlsConfig tlsConfig) {
+    super(config, routingManager, accessControlFactory, queryQuotaManager, tableCache, brokerMetrics);
+    _grpcConfig = buildGrpcQueryClientConfig(config);
+
+    // parse and fill constants
+    _maxBacklogPerServer = Integer.parseInt(_config.getProperty("MAXIMUM_BACKLOG_PER_SERVER", "1000"));
+    _grpcBrokerThreadPoolSize = Integer.parseInt(_config.getProperty("GRPC_BROKER_THREAD_POOL_SIZE", "10"));
+
+    // create streaming query client
+    _streamingQueryClient = new PinotStreamingQueryClient(_grpcConfig);
+
+    // create streaming reduce service
+    _streamingReduceService = new StreamingReduceService(config);
+  }
+
+  @Override
+  public void start() {
+  }
+
+  @Override
+  public synchronized void shutDown() {
+    _streamingReduceService.shutDown();
+  }
+
+  @Override
+  protected BrokerResponseNative processBrokerRequest(long requestId, BrokerRequest originalBrokerRequest,
+      @Nullable BrokerRequest offlineBrokerRequest, @Nullable Map<ServerInstance, List<String>> offlineRoutingTable,
+      @Nullable BrokerRequest realtimeBrokerRequest, @Nullable Map<ServerInstance, List<String>> realtimeRoutingTable,
+      long timeoutMs, ServerStats serverStats, RequestStatistics requestStatistics)
+      throws Exception {
+    assert offlineBrokerRequest != null || realtimeBrokerRequest != null;
+
+    String rawTableName = TableNameBuilder.extractRawTableName(originalBrokerRequest.getQuerySource().getTableName());
+    Map<ServerRoutingInstance, Iterator<Server.ServerResponse>> responseMap = new HashMap<>();
+    // Making request to a streaming server response.
+
+    if (offlineBrokerRequest != null) {
+      // to offline servers.
+      assert offlineRoutingTable != null;
+      Map<ServerRoutingInstance, Iterator<Server.ServerResponse>> serverResponses =
+          streamingQueryToPinotServer(requestId, _brokerId, rawTableName, TableType.OFFLINE,
+              offlineBrokerRequest, offlineRoutingTable, timeoutMs, true, 1);
+      responseMap.putAll(serverResponses);
+    }
+    if (realtimeBrokerRequest != null) {
+      // to realtime servers.
+      assert realtimeRoutingTable != null;
+      Map<ServerRoutingInstance, Iterator<Server.ServerResponse>> serverResponses =
+          streamingQueryToPinotServer(requestId, _brokerId, rawTableName, TableType.REALTIME,
+              realtimeBrokerRequest, realtimeRoutingTable, timeoutMs, true, 1);
+      responseMap.putAll(serverResponses);
+    }
+    BrokerResponseNative brokerResponse = _streamingReduceService.reduceOnStreamResponse(
+        originalBrokerRequest, responseMap, timeoutMs, _brokerMetrics);
+    return brokerResponse;
+  }
+
+  /**
+   * Query pinot server for data table.
+   */
+  public Map<ServerRoutingInstance, Iterator<Server.ServerResponse>> streamingQueryToPinotServer(
+      final long requestId,
```

Comment:
（minor, format) Can we put them into fewer lines? Also we don't usually put `final` for method arguments